pymc_marketing/mmm/base.py

```diff
@@ -1293,6 +1295,130 @@ def plot_channel_contribution_share_hdi(
+        dataframe = self.compute_mean_contributions_over_time(
+            original_scale=original_scale
+        )
+
+        dataframe = self._process_decomposition_components(data=dataframe)
+        total_contribution = dataframe["contribution"].sum()
+
+        # Initialize the matplotlib figure and axis
+        fig, ax = plt.subplots(figsize=figsize, **kwargs)
+
+        # Initialize the starting point for the first bar
+        cumulative_contribution = 0
+
+        # Plot each bar with the updated order
+        for index, row in dataframe.iterrows():
+            # Choose the color based on the sign of the contribution
+            color = "lightblue" if row["contribution"] >= 0 else "salmon"
+
+            # For negative contributions, start the bar at the cumulative sum minus the contribution
+            bar_start = (
+                cumulative_contribution + row["contribution"]
+                if row["contribution"] < 0
+                else cumulative_contribution
+            )
+            ax.barh(row["component"], row["contribution"], left=bar_start, color=color)
+
+            # Only add to the cumulative sum if the contribution is positive
+            if row["contribution"] > 0:
+                cumulative_contribution += row["contribution"]
+
+            # Label positioning
+            label_pos = bar_start + (row["contribution"] / 2)
+            # Ensure that the label is always inside the bar for visibility
+            if row["contribution"] < 0:
+                label_pos = bar_start - (row["contribution"] / 2)
+
+            # Add labels on top of the bars for the contribution values and percentages
+            ax.text(
+                label_pos,
+                index,
+                f"{row['contribution']:,.0f}\n({row['percentage']:.1f}%)",
+                ha="center",
+                va="center",
+                color="black",
+                fontsize=10,
+            )
+
+        # Set the title and labels
+        ax.set_title("Response Decomposition Waterfall by Components")
+        ax.set_xlabel("Cumulative Contribution")
+        ax.set_ylabel("Components")
+
+        # Adjust x-axis to show the percentage
+        xticks = np.linspace(
+            0, total_contribution, num=11
+        )  # 10 equally spaced ticks from 0 to total
+        xticklabels = [
+            f"{(x/total_contribution)*100:.0f}%" for x in xticks
+        ]  # Convert to percentages
+        ax.set_xticks(xticks)
+        ax.set_xticklabels(xticklabels)
+
+        # Hide the right, top, and left spines for a cleaner look
+        ax.spines["right"].set_visible(False)
+        ax.spines["top"].set_visible(False)
+        ax.spines["left"].set_visible(False)
+
+        # Add labels on the left to identify the predictor channels, corresponding to the y-ticks
+        ax.set_yticks(np.arange(len(dataframe)))
+        ax.set_yticklabels(dataframe["component"])
+
+        plt.tight_layout()
```

Comment:
i think you can remove this  (I think it has been depreciates but I am not sure) and use 

```python
fig, ax = plt.subplots(figsize=figsize, layout-"constrained", **kwargs)
```